clarity/src/vm/types/serialization.rs

```diff
@@ -287,6 +291,132 @@ macro_rules! check_match {
     };
 }
 
+impl TypeSignature {
+    /// Return the maximum length of the consensus serialization of a
+    /// Clarity value of this type. The returned length *may* not fit
+    /// in in a Clarity buffer! For example, the maximum serialized
+    /// size of a `(buff 1024*1024)` is `1+1024*1024` because of the
+    /// type prefix byte. However, that is 1 byte larger than the maximum
+    /// buffer size in Clarity.
+    pub fn max_serialized_size(&self) -> Result<u32, CheckErrors> {
```

Comment:
Do you wanna add a runtime `assert!()` to `Value::serialize_write()` to verify that the resulting byte buffer does indeed fit into the size reported by this function?

Reply:
Added in d8f7628